sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -883,6 +883,129 @@ class Dataset[T] private[sql](
     println(showString(numRows, truncate, vertical))
   // scalastyle:on println
 
+  /**
+   * Gets top 20 rows of Dataset as string in a tabular form. Strings more than 20 characters
+   * will be truncated, and all cells will be aligned right.
+   *
+   * @group action
+   * @since 3.5.0
+   */
+  def getString(): String = getString(20)
+
+  /**
+   * Gets the Dataset in a tabular form. Strings more than 20 characters will be truncated,
+   * and all cells will be aligned right. For example:
+   * {{{
+   *   year  month AVG('Adj Close) MAX('Adj Close)
+   *   1980  12    0.503218        0.595103
+   *   1981  01    0.523289        0.570307
+   *   1982  02    0.436504        0.475256
+   *   1983  03    0.410516        0.442194
+   *   1984  04    0.450090        0.483521
+   * }}}
+   *
+   * @param numRows Number of rows to show
+   * @group action
+   * @since 3.5.0
+   */
+  def getString(numRows: Int): String = getString(numRows, truncate = true)
+
+  /**
+   * Gets the top 20 rows of Dataset as string in a tabular form.
+   *
+   * @param truncate Whether truncate long strings. If true, strings more than 20 characters will
+   *                 be truncated and all cells will be aligned right
+   * @group action
+   * @since 3.5.0
+   */
+  def getString(truncate: Boolean): String = getString(20, truncate)
+
+  /**
+   * Gets top 20 rows of Dataset as string in a tabular form.
+   *
+   * @param truncate Whether truncate long strings. If true, strings more than 20 characters will
+   *                 be truncated and all cells will be aligned right
+   * @group action
+   * @since 3.5.0
+   */
+  def getString(numRows: Int, truncate: Boolean): String = if (truncate) {
+    showString(numRows, truncate = 20)
+  } else {
+    showString(numRows, truncate = 0)
+  }
+
+  /**
+   * Gets the Dataset as string in a tabular form. For example:
+   * {{{
+   *   year  month AVG('Adj Close) MAX('Adj Close)
+   *   1980  12    0.503218        0.595103
+   *   1981  01    0.523289        0.570307
+   *   1982  02    0.436504        0.475256
+   *   1983  03    0.410516        0.442194
+   *   1984  04    0.450090        0.483521
+   * }}}
+   *
+   * @param numRows  Number of rows to show
+   * @param truncate If set to more than 0, truncates strings to `truncate` characters and
+   *                 all cells will be aligned right.
+   * @group action
```

Comment:
i am not sure what should be the group these apis come under.  this wont be an action but not sure what is the group i need to add